### PR TITLE
Reduce shared_ptr copying in LogicalPlanner::plan and CallbackSink

### PR DESCRIPTION
**Pull request by mapleFU**

### CODE REVIEW

File: velox/exec/CallbackSink.h [hunk 1]
```diff
@@ -27,11 +27,11 @@ class CallbackSink : public Operator {
       DriverCtx* driverCtx,
       std::function<BlockingReason(RowVectorPtr, ContinueFuture*)> callback)
       : Operator(driverCtx, nullptr, operatorId, "N/A", "CallbackSink"),
-        callback_{callback} {}
+        callback_{std::move(callback)} {}
 
   void addInput(RowVectorPtr input) override {
     loadColumns(input, *operatorCtx_->execCtx());
-    blockingReason_ = callback_(input, &future_);
+    blockingReason_ = callback_(std::move(input), &future_);
```
Comment (mapleFU) [inline velox/exec/CallbackSink.h:34]: Here `callback_` copying argument will copying `input` once

File: velox/exec/LocalPlanner.cpp [hunk 2]
```diff
@@ -88,7 +89,7 @@ OperatorSupplier makeConsumerSupplier(
 
       auto consumer = [mergeSource](
                           RowVectorPtr input, ContinueFuture* future) {
-        return mergeSource->enqueue(input, future);
+        return mergeSource->enqueue(std::move(input), future);
```
Comment (mapleFU) [inline velox/exec/LocalPlanner.cpp:92]: And enqueue might also call copying once

**Comment by mapleFU** [comment]
@mbasmanova This patch minor optimize the shared_ptr usage in `CallbackSink` during building the operator and using it. Other usage like `dynamic_cast_pointer` might copying but they just copying in the `LogicalPlanner` and doesn't copy the data during exec

File: velox/exec/LocalPlanner.cpp [hunk 3]
```diff
@@ -52,7 +52,7 @@ namespace detail {
 
 /// Returns true if source nodes must run in a separate pipeline.
 bool mustStartNewPipeline(
-    std::shared_ptr<const core::PlanNode> planNode,
+    const std::shared_ptr<const core::PlanNode>& planNode,
```
Comment (mbasmanova) [inline velox/exec/LocalPlanner.cpp:55]: nit: might be good to change to core::PlanNodePtr

**Review by mbasmanova — APPROVED** [review]
Thanks.

Please, check PR titles for committed PRs to notice that they do not use "major/minor" and typically start with an active verb.

**Pull request closed without merging**